xrpl/utils/tx_parser/utils.py

```diff
@@ -0,0 +1,1052 @@
+"""Utils for balance_changes and orderbook_changes."""
+from __future__ import annotations
+
+from datetime import datetime
+from decimal import Decimal
+from typing import Any, Callable, Dict, List, Optional, Tuple, Type, Union
+
+from pydash import compact, filter_, flatten, group_by, map_, map_values  # type: ignore
+from typing_extensions import Literal
+
+from xrpl.constants import XRPLException
+from xrpl.utils.xrp_conversions import drops_to_xrp
+
+"""
+###########################################
+Utils balance_parser and orderbook_parser
+###########################################
+"""
+
+
+# metadata received from a tx method
+METADATA_TYPE = Dict[
+    str,
+    Union[
+        str,
+        int,
+        Dict[str, str],  # issued currency amount
+        List[List[Dict[str, Union[str, int]]]],  # Field: 'paths'
+        Dict[  # Field: 'meta'
+            str,  # 'AffectedNodes'
+            List[
+                Dict[
+                    str,  # Node state
+                    Dict[
+                        str,
+                        Union[str, int, Dict[str, Union[str, int, Dict[str, str]]]],
+                    ],
+                ],
+            ],
+        ],
+    ],
+]
+
+
+# metadata received from a subscribtion method
+SUBSCRIPTION_METADATA_TYPE = Dict[
+    str,
+    Union[
+        str,
+        int,
+        Dict[  # Field: 'transaction'
+            str,
+            Union[
+                str,
+                int,
+                Dict[str, str],  # issued currency amount
+                List[List[Dict[str, Union[str, int]]]],  # Field: 'paths'
+            ],
+        ],
+        Dict[  # Field: 'meta'
+            str,  # 'AffectedNodes'
+            List[
+                Dict[
+                    str,  # Node state
+                    Dict[
+                        str,
+                        Union[str, int, Dict[str, Union[str, int, Dict[str, str]]]],
+                    ],
+                ],
+            ],
+        ],
+    ],
+]
+
+
+class XRPLMetadataException(XRPLException):
+    """Exception for invalid transaction metadata."""
+
+    pass
+
+
+def is_valid_metadata(
+    metadata: Union[METADATA_TYPE, SUBSCRIPTION_METADATA_TYPE],
+) -> None:
+    """Check if the given metadata is valid.
+
+    Args:
+        metadata (Union[METADATA_TYPE, SUBSCRIPTION_METADATA_TYPE]):
+            The transactions metadata
+
+    Raises:
+        XRPLMetadataException: If field Account is missing.
+        XRPLMetadataException: If field meta is missing.
+        XRPLMetadataException: If field AffectedNodes is missing or is empty.
+    """
+    if "transaction" in metadata:
+        if "Account" not in metadata["transaction"]:  # type: ignore
+            raise XRPLMetadataException(
+                "Metadata incomplete: Metadata field 'Account' must be included."
```

Comment:
```suggestion
                "Malformed metadata: Metadata field 'Account' must be included."
```
ditto to the other messages